src/components/main-button.tsx

```diff
@@ -1,15 +1,41 @@
+// MainButton.tsx
 import React from 'react';
-import {Button, View} from 'react-native';
+import {TouchableOpacity, Text, StyleSheet} from 'react-native';
 
 interface MainButtonProps {
   title: string;
   onPress: () => void;
+  disabled?: boolean;
 }
 
-export const MainButton: React.FC<MainButtonProps> = ({title, onPress}) => {
+export const MainButton: React.FC<MainButtonProps> = ({
+  title,
+  onPress,
+  disabled = false,
+}) => {
```

Comment:
Nit: (opinião pessoal, não existe um padrão ou regra na empresa)

```suggestion
export const MainButton: React.FC<MainButtonProps> = (props: MainButtonProps) => {
  const { title, onPress, disabled = false } = props;
```

Normalmente quebrar o protótipo de uma função pode dificultar a leitura. Fazer o destruct na primeira linha da implementação tem o mesmo efeito.

Reply:
eu deixei numa linha só foi o lint que quebrou, vou mudar

Reply:
Se o lint quebrou é porque você desrespeitou o máximo de linhas na formatação. O ponto não é ter todo o protótipo  da função numa linha só em qualquer condição, acredito que tenham cenários em que isso não vai ser possível, dependendo do tamanho do nome e dos argumentos da função.

É importante também seguir o máximo de caracteres por linha recomendados do lint.

O ponto era justamente diminuir o tamanho do protótipo, e fazer o destruct das variáveis dentro da função para melhorar a legibilidade. Faz sentido?